test/api/simple_test.py

```diff
@@ -256,6 +257,41 @@ def test__api__config_path():
     assert parsed == expected_parsed
 
 
+@pytest.mark.parametrize(
+    "exclude_sqlfluff,kwargs,expected",
+    [
+        (
+            # No override from API, so uses .sqlfluff value
+            "L027,L029",
+            {},
+            set(),
+        ),
+        (
+            # API overrides, so it uses that
+            "L027,L029",
+            dict(exclude_rules=["L027"]),
+            {"L029"},
+        ),
+    ],
+)
+def test__api__config_override(exclude_sqlfluff, kwargs, expected, tmpdir):
+    """Test that parameters to lint() override .sqlfluff correctly (or not)."""
+    tmp_path = pathlib.Path(str(tmpdir))
+    config_path = tmp_path / ".sqlfluff"
+    with tmpdir.as_cwd():
+        config_path.write_text(
+            """[sqlfluff]
+exclude_rules = {}""".format(
+                exclude_sqlfluff
+            )
+        )
+        sql = "SELECT TRIM(name) AS name FROM some_table"
+        lint_results = sqlfluff.lint(sql, config_path=str(config_path), **kwargs)
+        assert expected == {"L027", "L029"}.intersection(
+            {lr["code"] for lr in lint_results}
+        )
```

Comment:
Rather than writing to a temp dir (which sounds flakey, prone to conflcits, and not very scalable) could we not just have a test `.sqlfluff` config file in our code in our `tests` dir and use that? I believe we use that model elsewhere.

Reply:
This test is also a little unusual in that it is made up of both SQL and config files but also the parameters that will be passed to the simple API (note that `expected` is a Python dictionary that we pass to a function).

Reply:
I’m more concerned with two tests running at the same time, and getting some sort of race condition as the file is written by one process and read by another.

This could even be within the one run if using our parallel processing technique.

`tmpdir` is an OS-agnostic way of getting access to the temp directory location, but it doesn’t create a unique dir/file within that AFAIK to avoid that situation.

Maybe we don’t need a new file for this and can just use one of the existing ones to avoid bloating the code base?

Reply:
Ah. This isn't the same tmpdir you're thinking of. This is a purest "fixture" that creates a unique directory every time (it'll be a subdirectory of the "main" temp directory). So it's guaranteed safe from those kinds of race conditions.

You can see the docs here. PyTest has a bunch of nice built-in features like this that make life easier.

https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture

Reply:
Ok, moved the `.sqlfluff` file to a directory in git -- no temp directories here. 🙏